packages/react-native-bpk-component-dialog/src/NativeDialog.ios.js

```diff
@@ -0,0 +1,120 @@
+/*
+ * Backpack - Skyscanner's Design System
+ *
+ * Copyright 2016-2019 Skyscanner Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/* @flow */
+
+import React, { Component } from 'react';
+import {
+  requireNativeComponent,
+  NativeModules,
+  NativeEventEmitter,
+} from 'react-native';
+import type EmitterSubscription from 'react-native/Libraries/vendor/emitter/EmitterSubscription';
+import isNil from 'lodash/isNil';
+
+import {
+  commonPropTypes,
+  // commonDefaultProps,
+  type CommonProps,
+} from './common-types';
+
+const RCTBPKDialog = requireNativeComponent('RCTBPKDialog');
+
+let uniqueModalIdentifier = 0;
+
+const BpkDialogEventEmitter = NativeModules.BPKDialogEventsManager
+  ? new NativeEventEmitter(NativeModules.BPKDialogEventsManager)
+  : null;
+
+export type Props = {
+  ...$Exact<CommonProps>,
+};
+
+export type State = {
+  ...$Exact<CommonProps>,
+};
+
+class BpkDialog extends Component<Props> {
+  eventSubscription: ?EmitterSubscription;
+
+  identifier: number;
+
+  constructor(props: Props) {
+    super(props);
+    this.identifier = uniqueModalIdentifier;
+    uniqueModalIdentifier += 1;
+  }
+
+  componentDidMount() {
+    if (BpkDialogEventEmitter) {
+      const { actions, scrimAction } = this.props;
+      this.eventSubscription = BpkDialogEventEmitter.addListener(
```

Comment:
I'd be interested to know the rationale behind using event emitters, instead of allowing regular RN Button elements to be passed to the component. I'm sure there's a good reason.

Reply:
I used the event emitters because is how RN does it with the Modal:
https://github.com/facebook/react-native/blob/9bb5c32de9ef06942e38ea6cf5d9000c37741c37/Libraries/Modal/Modal.js#L180-L191